checkov/sast/checks/checks_parser.py

```diff
@@ -0,0 +1,137 @@
+
+import logging
+from typing import Dict, Any, List
+
+from checkov.common.graph.checks_infra.enums import Operators
+from checkov.sast.consts import COMPARISON_VALUES, COMPARISON_VALUE_TO_SYMBOL, SemgrepAttribute, \
+    CHECKOV_SEVERITY_TO_SEMGREP_SEVERITY, PATTERN_OPERATOR_TO_SEMGREP_ATTR, VARIABLE_OPERATOR_TO_SEMGREP_ATTR, \
+    FILTER_OPERATOR_TO_SEMGREP_ATTR, BqlConditionType
+from checkov.common.util.type_forcers import force_list
+
+
+class SastCheckParser:
+    def parse_raw_check_to_semgrep(self, raw_check: Dict[str, Dict[str, Any]]) -> Dict[str, Any]:
+        semgrep_rule: Dict[str, Any] = {}
+        if not self.raw_check_is_valid(raw_check):
+            logging.error(f'cant parse the following policy: {raw_check}')
+        try:
+            semgrep_rule = self.parse_rule_metadata(raw_check, semgrep_rule)
+            semgrep_rule.update(self.parse_definition(raw_check['definition']))
+        except Exception as e:
+            raise e
+
+        return semgrep_rule
+
+    def raw_check_is_valid(self, raw_check: Dict[str, Any]) -> bool:
+        metadata = raw_check.get('metadata')
+        if not metadata:
+            raise AttributeError('BQL policy is missing the metadata field')
+        if not metadata.get('id'):
+            raise AttributeError('BQL policy metadata is missing an id value')
+        if not metadata.get('severity'):
+            raise AttributeError('BQL policy metadata is missing a severity')
+        if not raw_check.get('scope', {}).get('languages'):
+            raise AttributeError('BQL policy metadata is missing languages')
+
+        else:
+            return True
+
+    def parse_rule_metadata(self, bql_policy: Dict[str, Any], semgrep_rule: Dict[str, Any]) \
+            -> Dict[str, Any]:
+        metadata = bql_policy['metadata']
+        semgrep_rule[SemgrepAttribute.ID.value] = metadata['id']
+        semgrep_rule[SemgrepAttribute.MESSAGE.value] = metadata.get('guidelines', '')
+        semgrep_rule[SemgrepAttribute.SEVERITY.value] = CHECKOV_SEVERITY_TO_SEMGREP_SEVERITY[metadata['severity']]
+        languages = bql_policy['scope']['languages']
+        semgrep_rule[SemgrepAttribute.LANGUAGES.value] = languages
+        semgrep_rule['metadata'] = {'name': metadata['name']}
+
+        # add optional cwe and owasp metadata
+        cwe = metadata.get('cwe')
+        if cwe:
+            semgrep_rule['metadata'][SemgrepAttribute.CWE.value] = cwe
+        owasp = metadata.get('owasp')
+        if owasp:
+            semgrep_rule['metadata'][SemgrepAttribute.OWASP.value] = owasp
+
+        return semgrep_rule
+
+    def parse_definition(self, definition: Dict[str, Any]) -> Dict[str, Any]:
+        definitions = force_list(definition)
+        conf = {}
+        if len(definitions) > 1:
+            conf[SemgrepAttribute.PATTERNS.value] = self.get_definitions_list_items(definitions)
+        else:  # since definitions has only one value, we check only definitions[0]
```

Comment:
can it be 0 and then raise an `IndexOutOfBoundsError`?

Reply:
I don't think of an option where a definition is passed empty, but added a condition to make sure its not the case